classes/helper.php

```diff
@@ -155,11 +131,18 @@ public static function get_pdf_filerecord(string $filename, string $converter) :
      *
      * @param \moodle_url $targeturl the target URL to reach after passing through proxy.
      * @param string $key the access key to use for Moodle user login validation.
+     * @param int $instanceid the instance ID to use for access key. {@see \tool_pdfpages\key_manager::get_instance_id}.
      *
      * @return \moodle_url
      */
-    public static function get_proxy_url(moodle_url $targeturl, string $key) {
-        return new moodle_url('/admin/tool/pdfpages/index.php', ['url' => $targeturl->out(), 'key' => $key]);
+    public static function get_proxy_url(moodle_url $targeturl, string $key, int $instanceid) {
+        $params = [
+            'url' => $targeturl->out(),
+            'key' => $key,
+            'instanceid' => $instanceid
```

Comment:
proxy don't need this as long if it knows URL. Then key manager can check by url

Reply:
Removed from proxy as abstracted to use URL